shared/src/main/java/uk/gov/di/authentication/shared/state/UserContext.java

```diff
@@ -45,6 +49,10 @@ public Session getSession() {
         return session;
     }
 
+    public AuthSessionItem getAuthSession() {
+        return authSession.orElseThrow(() -> new RuntimeException("Auth session not set."));
```

Comment:
This is a good question. I actually made a mistake and was calling `buildUserContext` with a `null` auth session from the start handler instead of `optional.empty` (fixed now). From the other handlers, it will always be present as the base frontend handler returns a http error if the auth session isn't present. The reason for making it optional is really just a stopgap, as if you look in the main handle method, the auth session is constructed with  _upliftRequired_ as a parameter, but  _upliftRequired_ is computed off of the user context, which otherwise requires an auth session. The construction loop only effects this handler and having discussed it with Bill we thought it's something to be considered in a future refactor.